internal/tenant/tenant.go

```diff
@@ -62,3 +63,38 @@ func (t tenantService) GetTenantInfoByNamespace(apiURL string, ns string) (InfoL
 
 	return tenantInfo, nil
 }
+
+// returns true if the cluster the ns is on has reached maximum capacity
+func (t tenantService) HasReachedMaxCapacity(apiURL, ns string) (bool, error) {
+
+	ti, err := t.GetTenantInfoByNamespace(apiURL, ns)
+	if err != nil {
+		return true, err
+	}
+
+	if len(ti.Errors) != 0 {
+		firstError := ti.Errors[0]
+		err = fmt.Errorf("%s - %s", firstError.Code, firstError.Detail)
+		return true, err
+	}
+
+	namespaces := ti.Data[0].Attributes.Namespaces
+	index, found := indexOfNamespaceWithName(namespaces, ns)
+	if !found {
+		return true, fmt.Errorf("Failed to find %s in tenant info", ns)
+	}
+
+	jenkins := namespaces[index]
+	return jenkins.ClusterFull, nil
+}
+
+// returns the index of the namespace that equals 'name'
+func indexOfNamespaceWithName(namespaces []Namespace, name string) (int, bool) {
```

Comment:
nit (feel free to ignore): do we need to return both, an int and a bool, here? How about returning just a `-1` in the end which would convey that no match was found.

Reply:
could be ... but aligns with how `go` does this with maps  `[ ]` operator 

```
v, ok :=  lookup[key]
```

but in that case, there is a good reason to return bool. I will change the implementation.